README.md

```diff
@@ -65,8 +65,6 @@ For integration with other platform specifications, having the ability to refere
 
 Making it a static method on `String` seems like the safest home for such a method. Adding custom methods to `String.prototype` is likely quite risky, but could be considered as well.
 
-### Are the primary benefits performance?
+### What about performance?
 
-While the proposal is not entirely for performance reasons, it should hopefully still enable a builtin method that could be faster than user validation.
-
-In future it might even be possible for a bit state to be associated with string data types and maintained through string functions to make the check entirely zero cost, but that would be entirely up to implementations and not anything within the reach of this specification.
+Performance optimizations are up to implementations and are not the primary motivation for this specification, yet enabling a builtin method that is faster than validation in userland is desireable. Perhaps well-formedness state could be cached per string and propagated through string operations so common cases to determine well-formedness do not require a linear scan, say with two bits indicating well-formedness (00<sub>2</sub>), low surrogate at the start and/or high surrogate at the end (01<sub>2</sub>), unpaired surrogate somewhere in the middle (10<sub>2</sub>) and deferred rescan (11<sub>2</sub>).
```

Comment:
I'd prefer the discussion here to avoid too much of the implementation detail and just focus on the high level, perhaps something like:

```suggestion
Performance optimizations are up to implementations and are not the primary motivation for this specification, yet enabling a builtin method that is faster than validation in userland is desireable. Perhaps well-formedness state could be cached per string and propagated through string operations to avoid the need for a linear scan since the concatenation of well formed strings is a well formed string and slices of well formed strings only require checking on the boundaries.
```

Reply:
What I am trying to convey is that optimization might not be as trivial as one might assume, at least in certain cases. Perhaps:

```suggestion
Performance optimizations are up to implementations and are not the primary motivation for this specification, yet enabling a builtin method that is faster than validation in userland is desirable. Perhaps well-formedness state could be cached per string and propagated through string operations to avoid the need for a linear scan where possible, say where strings are already well-formed, while handling of strings containing inner unpaired surrogates may still require linear-time (re)scans.
```